src/lib/y2partitioner/actions/create_partition_table.rb

```diff
@@ -20,74 +20,72 @@
-    # visual glitches, like a completely empty wizard step (in the DASD case).
-    class CreatePartitionTable < TransactionWizard
-      attr_reader :controller
-
-      # @param disk_name [String]
-      def initialize(disk_name)
+    class CreatePartitionTable
+      include Yast::I18n
+      include Yast::Logger
+      include ConfirmRecursiveDelete
+
+      # @param disk [Y2Storage::Partitionable]
+      def initialize(disk)
         textdomain "storage"
+        @disk = disk
+      end
 
-        super()
-        @controller = Controllers::PartitionTable.new(disk_name)
+      # Checks whether the action can be performed, displays a confirmation
+      # popup and, if everything goes fine so far, performs the action.
+      #
+      # @return [Symbol]
+      def run
+        return :back unless validate && confirm
+
+        perform_action
+        :finish
       end
 
     protected
 
-      # @see TransactionWizard
-      def sequence_hash
-        {
-          "ws_start"    => "select_type",
-          "select_type" => { next: "confirm" },
-          "confirm"     => { next: "commit" },
-          "commit"      => { finish: :finish }
-        }
+      # @return [Y2Storage::Partitionable]
+      attr_reader :disk
+
+      # Device name of the partitionable device
+      # @return [String]
+      def disk_name
+        disk.name
       end
 
-      skip_stack :select_type
+      # Creates the new partition table, asking the user for the type if needed
+      #
+      # Does nothing if the user cancels the action when asked for the type
+      def perform_action
+        type = possible_types.size > 1 ? selected_type : default_type
+        return if type.nil?
 
-      # Open a dialog to let the user select the partition table type
-      # if there is more than one to select from.
-      def select_type
-        return :next unless controller.multiple_types?
-        Dialogs::PartitionTableType.run(controller)
+        create_partition_table(type)
       end
 
-      # Ask the user for confirmation message before creating the partition
-      # table.
-      def confirm
-        # TRANSLATORS %s is the disk device name ("/dev/sda" or similar)
-        msg = _("Really create a new partition table on %s?") % disk_name
-        msg += "\n\n"
-        msg += _("This will delete all existing partitions on that device\n" \
-                 "and all devices (LVM volume groups, RAIDs etc.)\n" \
-                 "that use any of those partitions!")
-        Yast::Popup.YesNo(msg) ? :next : :back
-      end
-
-      # Commit the action (creating a new partition table) to the staging
-      # devicegraph. This does not open a dialog.
-      def commit
-        log.info("Commit creating a new #{controller.type} partition table on #{disk_name}")
-        controller.create_partition_table
-        :finish
+      # Creates the new partition table
+      #
+      # @param type [Y2Storage::PartitionTables::Type]
+      def create_partition_table(type)
```

Comment:
this method looks a bit like it should be in Partitionable itself. why original one does not remove descendants?

Reply:
That's by design of libstorage-ng. Usually the methods there are kind of low-level with very little "implicit" or "automagical" behavior. The user of the library is expected to do the preparations explicitly if needed.

And the `Y2Storage` wrappers around the `Storage` classes try to respect as much as possible the original methods and ideas.

Reply:
so what about adding something like "safe" variant or having convention that original ones that is rough ends up "!" and new safer ones will be without it? I just think that partitioner is not ideal place for it.

Reply:
To avoid becoming inconsistent, we would need to do that for many libstorage-ng methods, not only `create_partition_table`. That is, renaming a bunch of libstorage-ng methods to `create_xxx!` or 'add_yyy!` and redefining the corresponding `create_xxx` and `add_yyy` to work in a non-libstorage way.

Sounds too aggressive to me (i.e. against the philosophy of keeping the wrapper thin). The whole libstorage-ng approach/API can be considered quite low-level not only in cases like this, but also in many other aspects. Once you change this in the wrapper, you will want to also change another behavior somewhere to make it also "smarter", and then another... where to stop?